langchain/src/vectorstores/azuresearch.ts

```diff
@@ -0,0 +1,541 @@
+        fields: [DEFAULT_FIELD_CONTENT_VECTOR],
+        kNearestNeighborsCount: k,
+      }],
+      filter,
+      top: k,
+    });
+
+    const docsWithScore: [Document, number][] = [];
+
+    for await (const item of results) {
+      const document = new Document({
+        pageContent: item.document.content,
+        metadata: item.document.metadata,
+      });
+
+      docsWithScore.push([document, item.score]);
+    }
+
+    return docsWithScore;
+  }
+
+  /**
+   * Perform a hybrid search using Semantic configuration.
+   *
+   * @param query
+   * @param queryVectors
+   * @param k
+   * @param filter
+   */
+  async semanticHybridSearchVectorWithScore(
+    query: string,
+    queryVectors: number[],
+    k: number,
+    filter?: string
+  ): Promise<[Document, number][]> {
+    const { results } = await this.client.search(query, {
+      vectors: [{
+        value: queryVectors,
+        fields: [DEFAULT_FIELD_CONTENT_VECTOR],
+        kNearestNeighborsCount: k,
+      }],
+      filter,
+      top: k,
+      queryType: "semantic",
+      queryLanguage: this.params.search.language ?? "en-us",
+      semanticConfiguration: this.params.search.semantic ?? "default",
+      captions: "extractive",
+      answers: "extractive",
+    });
+
+    const docsWithScore: [Document, number][] = [];
+
+    for await (const item of results) {
+      const document = new Document({
+        pageContent: item.document.content,
+        metadata: item.document.metadata,
+      });
+
+      docsWithScore.push([document, item.rerankerScore ?? item.score]);
+    }
+
+    return docsWithScore;
+  }
+
+  /**
+   * Execute a vector similarity search.
+   *
+   * @param query
+   * @param k
+   * @param filter
+   */
+  async similaritySearchVectorWithScore(
+    query: number[],
+    k: number,
+    filter?: string
+  ): Promise<[Document, number][]> {
+    const { results } = await this.client.search("", {
+      vectors: [{
+        value: query,
+        fields: [DEFAULT_FIELD_CONTENT_VECTOR],
+        kNearestNeighborsCount: k,
+      }],
+      filter,
+    });
+
+    const docsWithScore: [Document, number][] = [];
+
+    for await (const item of results) {
+      const document = new Document({
+        pageContent: item.document.content,
+        metadata: item.document.metadata,
+      });
+
+      docsWithScore.push([document, item.score]);
+    }
+
+    return docsWithScore;
+  }
+
+  /**
+   * Ensure that index exists on Vector Store.
+   *
+   * @param dbConfig
+   * @protected
+   */
+  protected static async ensureIndexExists(dbConfig: AzureSearchStoreParams): Promise<SearchClient<AzureSearchDocument>> {
+    if (isSearchClient(dbConfig.client)) return dbConfig.client;
+
+    const {indexName, endpoint} = dbConfig.client;
+    const credential = new AzureKeyCredential(dbConfig.client.credential);
+
+    const indexClient = new SearchIndexClient(endpoint, credential);
+
+    try {
+      await indexClient.getIndex(indexName);
```

Comment:
It seems that even getIndex requires an admin key and will fail with a query key. So you pass in a query key, exception gets thrown, after which the createIndex call will also fail

Reply:
Yeah, the implementation requires an Admin Key, but if you want to use the query key you can also pass the SearchClient instead of indexName, doing that I'll not validate if index exists and all works.

That is the example:

```typescript
AzureSearchStore.create({
    client: new SearchClient(...),
    search: {...}
});
```

I think is nice to have that in the documentation and some test's to validate that approach. Make sense?